ListLinked/LinkedList.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace LinkedList
+{
+    public  class LinkedList<T> : IEnumerable<T>
+    {
+        private int position = 0;
+        public Node<T> head;
+
+
+        public void Add(T value)
+        {
+            if (head == null)
+            {
+                head = new Node<T>(value);
+                return;
+            }
+            Node<T> current = head;
+            while (current.next != null)
+            {
+                current = current.next;
+            }
+            current.next = new Node<T>(value);
+
+        }
+
+        public void AddAt(int index, T value)
+        {
+            Node<T> current = head;
+            int head_position = 0;
+            while (head_position < index - 1)
+            {
+                current = current.next;
+                head_position++;
+            }
+            var next = current.next;
+            current.next = new Node<T>(value);
+            current.next.next = next;
+        }
+
+        public int Length()
```

Comment:
That one of the ways to implement it. Another one is to create a local field _length and increment/decrement it when you add/remove items from the list. Which of the approaches is better from the performance point of view?